File: First_repeating_element/code.cpp
class Solution {
  public:
    // Function to return the position of the first repeating element.
    int firstRepeated(int arr[], int n) {
        // code here
        map<int, int> check;
        for(int i=0; i<n; i++){
            check[arr[i]]++;
        }
        for(int i=0; i<n; i++){
            if(check[arr[i]] > 1){
                return i+1;
            }
        }
        return -1;
    }
};

File: First_repeating_element/code.java
class Solution {
    
    public static int firstRepeated(int[] arr, int n) {
        Map<Integer, Integer> mp = new HashMap<>();
        int count = 0;
        for(int i=0; i<n; i++){
            if(!mp.containsKey(arr[i])){
                mp.put(arr[i], 1);
            }
            else{
                mp.put(arr[i], mp.get(arr[i])+1);
            }
        }
        for(int j=0; j<n; j++){
            count++;
            if(mp.get(arr[j])>1)
                return count;
        }
        return -1;
    }
}
